src/coreclr/interpreter/compiler.cpp

```diff
@@ -5161,12 +5181,16 @@ void InterpCompiler::GenerateCode(CORINFO_METHOD_INFO* methodInfo)
             }
 
             case CEE_ISINST:
+            case CEE_CASTCLASS:
             {
                 CHECK_STACK(1);
                 CORINFO_RESOLVED_TOKEN resolvedToken;
                 ResolveToken(getU4LittleEndian(m_ip + 1), CORINFO_TOKENKIND_Casting, &resolvedToken);
 
-                CorInfoHelpFunc castingHelper = m_compHnd->getCastingHelper(&resolvedToken, false /* throwing */);
+                bool isCastClass = (opcode == CEE_CASTCLASS);
```

Comment:
It seems that we should also change the `PushInterpType` argument to `InterpTypeO` for the castclass type.